src/MxcImageProvider.cpp

```diff
@@ -108,6 +111,12 @@ MxcImageProvider::download(const QString &id,
     if (temp != infos.end())
         encryptionInfo = *temp;
 
+    if (cachedThumbnails.contains(id))
+    {
+        then(id, requestedSize, *cachedThumbnails[id], QString{});
```

Comment:
I wonder if leaving out the filepath will cause issues.

Reply:
Yeah, I'm not sure about that either. Think I should cache the thumbnail path as well?